subprojects/docs/src/docs/userguide/dependencyLocking.adoc

```diff
@@ -0,0 +1,152 @@
+    <sourcefile file="gradle/dependency-locks/compileClasspath.lockfile"/>
+</sample>
+++++
+
+which matches the following dependency declaration:
+
+++++
+<sample id="locking-one" dir="userguide/dependencyManagement/dependencyLocking/lockingSingleConfiguration" title="Dynamic dependency declaration">
+    <sourcefile file="build.gradle" snippet="locking-deps"/>
+</sample>
+++++
+
+=== Running a build with a lockfile present
+
+The moment a build needs to resolve a configuration that has locking enabled and it finds a matching lockfile,
+it will use it to verify that the given configuration still resolves the same versions.
+The validation is as follows:
+
+* Existing entries in the lockfile must be matched in the build
+** A version mismatch or missing resolved module causes a build failure
+* Resolution result must not contain extra dependencies compared to the lock file content
+
+=== Selectively updating lock file entries
+
+In order to update only specific modules of a configuration, you can use the `--update-locks` command line flag.
+It takes a comma (`,`) separated list of module notations.
+In this mode, the existing lock file is still used as input to resolution, filtering out the modules targeted by the update.
+
+----
+❯ gradle classes --update-locks org.apache.commons:commons-lang3,org.slf4j:slf4j-api
+----
+
+Wildcards, indicated with `*`, can be used in the group or module name. They can be the only character or appear at the end of the group or module respectively.
+The following wildcard notation examples are valid:
+
+* `org.apache.commons:*`: will let all modules belonging to group `org.apache.commons` update
+* `*:guava`: will let all modules named `guava`, whatever their group, update
+* `org.springframework.spring*:spring*`: will let all modules having their group starting with `org.springframework.spring` and name starting with `spring` update
+
+[NOTE]
+====
+The resolution may cause other module versions to update, as dictated by the Gradle resolution rules.
+====
+
+=== Disabling dependency locking
+
+. Make sure that the configuration for which you no longer want locking is not configured with locking.
+. Remove the file matching the configurations where you no longer want locking.
+
+If you only perform the second step above, then locking will effectively no longer be applied.
+However, if that configuration happens to be resolved in the future at a time where lock state is persisted, it will once again be locked.
+
+=== Locking limitations
+
+* It is currently not possible to lock the <<sec:applying_plugins_buildscript,`classpath` configuration>> used for build plugins.
+* Locking can not be applied to source dependencies.
+
+=== Nebula locking plugin
+
+This feature is inspired by the https://github.com/nebula-plugins/gradle-dependency-lock-plugin[Nebula Gradle dependency lock plugin].
+
+If you were using that plugin and want to migrate, please see the Nebula documentation.
```

Comment:
I'd remove the word "please". It is generally considered needless politeness in technical writing. I learned that from professional technical writers I worked with a while ago. See also http://www.drdobbs.com/architecture-and-design/10-things-tech-writers-want-developers-t/240165685 and https://developers.google.com/style/tone#politeness-and-use-of-please.